mesonbuild/options.py

```diff
@@ -844,7 +844,9 @@ def get_pending_value(self, key: T.Union[OptionKey, str], default: T.Optional[El
         return self.pending_options.get(key, default)
 
     def get_value(self, key: T.Union[OptionKey, str]) -> ElementaryOptionValues:
-        return self.get_value_object(key).value
+        key = self.ensure_and_validate_key(key)
```

Comment:
Why are you doing the `ensure_and_validate` here, then calling into a function where you now call `ensure_and_validate`?